CMakeLists.txt

```diff
@@ -557,6 +560,22 @@ if (LBANN_WITH_VTUNE)
   endif (VTune_FOUND)
 endif (LBANN_WITH_VTUNE)
 
+if (LBANN_WITH_CALIPER)
+  find_package(caliper REQUIRED)
+  find_package(adiak REQUIRED)
+  if(caliper_FOUND AND adiak_FOUND)
+    set(LBANN_HAS_CALIPER TRUE)
+    message(STATUS "LBANN_WITH_CALIPER")
+    configure_file(
+      "${CMAKE_SOURCE_DIR}/cmake/configure_files/adiak_config.hpp.in"
+      "${CMAKE_BINARY_DIR}/adiak_config.hpp"
+      @ONLY)
+  else ()
+    set(LBANN_HAS_CALIPER FALSE)
+    set(LBANN_WITH_CALIPER OFF)
```

Comment:
Is this branch meaningful? `find_package(... REQUIRED)` should kill the build process if these are not found.